File: src/main/java/com/server/ordering/domain/dto/response/RestaurantInfoDto.java
package com.server.ordering.domain.dto.response;

import com.server.ordering.domain.FoodCategory;
import com.server.ordering.domain.OrderType;
import com.server.ordering.domain.Restaurant;
import com.server.ordering.domain.RestaurantType;
import lombok.*;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class RestaurantInfoDto {

    private String ownerName;
    private String restaurantName;
    private String address;
    private String notice;
    private RestaurantType restaurantType;
    private FoodCategory foodCategory;
    private Integer tableCount;
    private Integer orderingWaitingTime;
    private Integer admissionWaitingTime;
    private double latitude;
    private double longitude;

    public RestaurantInfoDto(Restaurant restaurant) {
        this.ownerName = restaurant.getOwnerName();
        this.restaurantName = restaurant.getRestaurantName();
        this.address = restaurant.getAddress();
        this.notice = restaurant.getNotice();
        this.restaurantType = restaurant.getRestaurantType();
        this.foodCategory = restaurant.getFoodCategory();
        this.tableCount = restaurant.getTableCount();
        this.orderingWaitingTime = restaurant.getOrderingWaitingTime();
        this.admissionWaitingTime = restaurant.getAdmissionWaitingTime();
        this.latitude = restaurant.getLocation().getY();
        this.longitude = restaurant.getLocation().getX();
    }
}

//대표자명, 음식점명, 주소, 주문타입, 음식타입, 테이블수, 웨이팅 대기시간, 주문 대기시간

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/response/RestaurantInfoDto.kt
package com.byeoru.ordering_server.domain.dto.response

import com.byeoru.ordering_server.domain.FoodCategory
import com.byeoru.ordering_server.domain.Restaurant
import com.byeoru.ordering_server.domain.RestaurantType

class RestaurantInfoDto(restaurant: Restaurant) {

    val ownerName: String
    val restaurantName: String
    val address: String
    val notice: String?
    val restaurantType: RestaurantType
    val foodCategory: FoodCategory
    val tableCount: Int?
    val orderingWaitingTime: Int
    val admissionWaitingTime: Int
    val latitude: Double
    val longitude: Double

    init {
        ownerName = restaurant.ownerName
        restaurantName = restaurant.restaurantName
        address = restaurant.address
        notice = restaurant.notice
        restaurantType = restaurant.restaurantType
        foodCategory = restaurant.foodCategory
        tableCount = restaurant.tableCount
        orderingWaitingTime = restaurant.orderingWaitingTime
        admissionWaitingTime = restaurant.admissionWaitingTime
        latitude = restaurant.location.y
        longitude = restaurant.location.x
    }
}